Simulator.lua

```diff
@@ -1134,6 +1137,42 @@ end
 
 
 
+-- Puts APIEndpoints in the sandbox if one of all provided patterns match it.
+function Simulator:prefillApi(a_ApiDesc)
+	-- Returns true if the provided string matches any of the prefillSymbolPatterns
+	local function matchesAnyPattern(a_Target)
+		for _, pattern in pairs(self.options.prefillSymbolPatterns) do
+			if (a_Target:match(pattern)) then
+				return true
+			end
+		end
+		return false;
+	end
+	
+	local function fillApi(a_ApiDesc, a_SymbolName, a_Sandbox)
+		for functionName, functionDesc in pairs(a_ApiDesc.Functions) do
```

Comment:
I mean that `fillApi` should check that its parameters are valid, just like all the other `Simulator`'s functions do. It helps catch errors early on.